scripts/pre_commit_linter_test.py

```diff
@@ -0,0 +1,283 @@
+        super(HTMLLintTests, self).setUp()
+        self.js_ts_lint_swap = self.swap(
+            pre_commit_linter.JsTsLintChecksManager, 'perform_all_lint_checks',
+            mock_perform_all_lint_checks)
+        self.check_codeowner_swap = self.swap(
+            pre_commit_linter, 'check_codeowner_file',
+            mock_check_codeowner_file)
+
+    def test_valid_html_file(self):
+        with self.print_swap, self.js_ts_lint_swap, self.check_codeowner_swap:
+            pre_commit_linter.main(args=['--path=%s' % VALID_HTML_FILEPATH])
+        self.assertTrue(all_checks_passed(self.linter_stdout))
+        self.assertTrue('SUCCESS   HTML linting passed' in self.linter_stdout)
+
+    def test_invalid_html_file(self):
+        with self.print_swap, self.sys_swap:
+            with self.js_ts_lint_swap, self.check_codeowner_swap:
+                pre_commit_linter.main(
+                    args=['--path=%s' % INVALID_HTML_FILEPATH])
+        self.assertFalse(all_checks_passed(self.linter_stdout))
+        self.assertTrue((
+            '%s: line 8, col 10, duplicate attribute: ng-controller\n\n'
+            '[htmllint] found 1 errors out of 1 files\n' % INVALID_HTML_FILEPATH
+            ) in self.linter_stdout)
+        self.assertTrue('FAILED   HTML linting failed' in self.linter_stdout)
+
+
+class CSSLintTests(LintTests):
+    """"Test the CSS lint functions."""
+    def setUp(self):
+        super(CSSLintTests, self).setUp()
+        self.js_ts_lint_swap = self.swap(
+            pre_commit_linter.JsTsLintChecksManager, 'perform_all_lint_checks',
+            mock_perform_all_lint_checks)
+        self.check_codeowner_swap = self.swap(
+            pre_commit_linter, 'check_codeowner_file',
+            mock_check_codeowner_file)
+
+    def test_valid_css_file(self):
+        with self.print_swap, self.js_ts_lint_swap, self.check_codeowner_swap:
+            pre_commit_linter.main(args=['--path=%s' % VALID_CSS_FILEPATH])
+        self.assertTrue(all_checks_passed(self.linter_stdout))
+        self.assertTrue(
+            appears_in_linter_stdout(
+                ['SUCCESS   1 CSS file linted'],
+                self.linter_stdout)
+            )
+
+    def test_invalid_css_file(self):
+        with self.print_swap, self.sys_swap:
+            with self.js_ts_lint_swap, self.check_codeowner_swap:
+                pre_commit_linter.main(
+                    args=['--path=%s' % INVALID_CSS_FILEPATH])
+        self.assertFalse(all_checks_passed(self.linter_stdout))
+        self.assertTrue(
+            appears_in_linter_stdout(
+                ['18:16',
+                 'Unexpected whitespace before \":\"   declaration-colon-spac'
+                 'e-before',
+                 'FAILED    1 CSS file'],
+                self.linter_stdout
+                )
+            )
+
+
+class PythonLintTests(LintTests):
+    """Test the Python lint functions."""
+    def setUp(self):
+        super(PythonLintTests, self).setUp()
```

Comment:
We have some custom pylint tests as well (`pylint_extensions.py`). So, in the current test you have one valid and one invalid file being checked for the rules defined in `pylintrc`. I would suggest that have one more valid and invalid file for any one of the extensions as well. Similar for other checks if they have extensions.

Reply:
Okay, done! Added for the pylint extension that checks for space after yield keyword.